reactive/kubernetes_master.py

```diff
@@ -1737,12 +1737,18 @@ def switch_auth_mode(forced=False):
 @when_not('kubernetes-master.pod-security-policy.applied')
 def create_pod_security_policy_resources():
     pod_security_policy_path = '/root/cdk/pod-security-policy.yaml'
-
-    render(
-        'rbac-pod-security-policy.yaml',
-        pod_security_policy_path,
-        {}
-    )
+    pod_security_policy = hookenv.config('pod-security-policy')
+    if pod_security_policy:
+        hookenv.log('Using configuration defined on pod-security-policy option')
+        write_file_with_autogenerated_header(pod_security_policy_path,
+                                             pod_security_policy)
```

Comment:
This is user input, I think some form of validation of the input should be done. Someone could get a lot of strange errors out of this if they poorly format it.

Reply:
There is no need for adding complexity in the validation of the policy, the policy is applied in line 1754 and any format/string error will be raised to the hookenv error log.

Reply:
I see what you're saying, but the use case has changed. Originally, the content was controlled by the charm and on a failure it was logged in the debug-log and retried. However, now that you're applying user input you need to surface a bad input to the user not just log it in the debug-log.

A user with a bad config would get log messages that it failed, but the charm itself isn't going to show any failures in `juju status` at least none that I see. Especially for something like a security policy I think the charm needs to provide feedback to the user since you're now accepting user input. Debug logging is fine for developers, which is where the default policy comes from.

Reply:
do you have any suggestion on how to implement such validation?